master/internal/restore.go

```diff
@@ -246,16 +234,17 @@ var experimentSnapshotShims = map[int]snapshotShimFunc{
 	1: shimExperimentSnapshotV1,
 	2: shimExperimentSnapshotV2,
 	4: shimExperimentSnapshotV4,
+	5: shimExperimentSnapshotV5,
 }
 
 // shimExperimentSnapshot shims an experiment snapshot to the version required by the master,
 // returning an error in the event the shim fails or the snapshot version is greater
 // than the current version (which could happen in a downgrade).
-func shimExperimentSnapshot(snapshot []byte, version int) ([]byte, error) {
-	return shimSnapshot(experimentSnapshotShims, snapshot, version)
+func shimExperimentSnapshot(experimentID int, snapshot []byte, version int) ([]byte, error) {
+	return shimSnapshot(experimentID, experimentSnapshotShims, snapshot, version)
 }
```

Comment:
question: why the extra helper function?  Why not just `shimSnapshot` directly, and reference `experimentShapshotShims` directly?

Kinda seems like there were dreams of having a generic shimming tool, but that seems kinda pointless, and this is just extra layers of indirection.